scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
```

Comment:
`pep8 scipy` (run on the top level of the scipy repository) with version 1.7.0 of the pep8 checker tool is required. You can get it via `pip install pep8==1.7.0`.